File: app/services/priority_service.py
def compute_priority(task, deadline, sender):
    score = 0

    # urgency keywords
    urgent_words = ["urgent", "asap", "important", "deadline"]

    if any(word in task.lower() for word in urgent_words):
        score += 5

    # deadline proximity
    if deadline:
        score += 3

    # sender importance (basic)
    if "professor" in sender.lower():
        score += 2

    return score

File: backend/src/services/priority.service.js
export function computePriority(task, deadline, sender) {
  let score = 0;
  const urgentWords = ['urgent', 'asap', 'important', 'deadline'];

  if (urgentWords.some(w => (task || '').toLowerCase().includes(w))) score += 5;
  if (deadline) score += 3;
  if ((sender || '').toLowerCase().includes('professor')) score += 2;

  return score;
}
